### PR TITLE
Save and Edit documents in My Orders panel

### LINKED ISSUE #1975: Save and Edit documents in `My Orders` panel
In the `PlanningChannel` we have the `My Orders` tab.  When the rows in this table are expanded, we use JSON-editor to display the contents.

While the orders are initially show read-only `[*]`, the owner of each message (`message.details.from.roleId === playerRoleId`) should have an `Edit button`.  They can then make document edits and then `Save` or `Cancel` the edit.

`[*]` `<JsonEditor>` is currently _ignoring_ the `disabled` property. Bug in #1973

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🧰 Issue

Fixes #1975 

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by lilitkarapetyan** [comment]
Hello Ian, I'm still working on this PR.

File: packages/components/src/local/organisms/planning-messages-list/index.tsx [hunk 1]
```diff
@@ -125,16 +104,39 @@ export const PlanningMessagesList: React.FC<PropTypes> = ({
...
+            const details: MessageDetails = {
+              channel: channel.uniqid,
+              from: {
+                force: selectedForce.name,
+                forceColor: selectedForce.color,
+                roleName: selectedRoleName,
+                roleId: playerRoleId,
+                iconURL: selectedForce.iconURL || selectedForce.icon || ''
+              },
+              messageType: template.title,
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-messages-list/index.tsx]: @lilitkarapetyan - I'd like use to store the template id in `messageType`, please.  That means Serge won't fail if the template title is changed.

File: packages/components/src/local/molecules/json-editor/index.tsx [hunk 2]
```diff
@@ -190,18 +209,31 @@ export const JsonEditor: React.FC<Props> = ({
...
+      if (formId && !editButton) {
+        editor.disable()
+      } else if (disabled && !formId) {
         editor.disable()
       } else {
         editor.enable()
       }
     }
-  }, [editor])
+  }, [editor, editButton])
```
Comment (IanMayo) [inline packages/components/src/local/molecules/json-editor/index.tsx:232]: @lilitkarapetyan - this method refers to `formId`.  Should it include `formId` in the hook props?

Comment (lilitkarapetyan) [reply packages/components/src/local/molecules/json-editor/index.tsx:232]: Yes, because only components containing SaveMessageButton have a formId

**Comment by IanMayo** [comment]
Hello @lilitkarapetyan - I've just been able to reproduce an issue I saw yesterday.

* yarn develop
* log in as `Mar-4`
* edit document owned by `Mar-4`
* change document `description` by adding a couple of words
* press `Save`
* data in table doesn't update

I checked the database, and the new record is in there.
I checked the console, and saw this:
image

So, it looks like we're receiving the new message twice, and ignoring the new version.

Note: I'm pretty sure I'm not just getting this on browser that does the `Save`. I think I get it on other logged in browsers, too.

Have you been able to successfully edit/save a document?

**Comment by IanMayo** [comment]
Another issue, please @lilitkarapetyan 

When we save a document from `My orders`, it increments the `.message.Reference`.  

I'm pretty sure that in `checkReference` we should only find/generate a reference number if the `Reference` field is something like `null` or `''`.  I don't think we can check for `undefined`, because I think that would generate a `Reference` for templates that don't include a `Reference` field.

File: packages/helpers/src/tests/handle-initial-channel-messages.spec.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ describe('handle initial channel creation', () => {
     // get the blue RFI channel
     const rfiChan = res.channels['channel-BlueRFI']
     expect(rfiChan).toBeTruthy()
```
Comment (IanMayo) [inline packages/helpers/src/tests/handle-initial-channel-messages.spec.ts:43]: thanks for the fixes @lilitkarapetyan 

_One day_ I look forward to `handle-initial-channel-messages` getting refactored & tidied up

File: packages/components/src/local/organisms/data-table/index.spec.tsx [hunk 4]
```diff
@@ -20,6 +20,7 @@ const rfiData = newest.map((message: any) => {
     messageItem.details.channel,
     messageItem.details.from.roleName,
     messageItem.details.from.forceColor,
+    messageItem.details.from.forceId,
```
Comment (IanMayo) [inline packages/components/src/local/organisms/data-table/index.spec.tsx:23]: this one was a weird one to diagnose. Thanks!

**Comment by IanMayo** [comment]
@lilitkarapetyan - there was an issue, where if we edit/save a message, the old and new instances of the message were displayed.  I've found the [fix](https://github.com/serge-web/serge/pull/1999/commits/f6376c0d8a5713d2a457d6cb64f40858a6ea37b2#diff-f97fc21549a839fe120adefe5de4fc716b655a6bcc52eb70d83868a04aaea726R42) for it, in `handle-channel-updates` - `handleNonInfoMessage`.  

The issue was because we're 1/2 through moving from `Reference` to `reference`.   Damn, I can't remember why I wanted to move to lower-case `reference`.  But, it's fortunate we did it, because currently `serge` is updating the `Reference` on every message sent, rather than only doing it for newly created messages.  I'll check we have that issue captured.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**